src/views/FoldersView.tsx

```diff
@@ -33,6 +33,10 @@ export function FoldersView({ onClick }: { onClick?: (fileEntry: FileEntry) => v
       readAllProjectFiles().then(setFiles);
       console.log(files);
     });
+    runPDFIngestion().then(() => {
+      console.log('PDFs ingested with success');
+      readAllProjectFiles().then(setFiles);
+    });
```

Comment:
@cguedes @sehyod Advice or thoughts on how to do this piece are much appreciated (my JS/TS/React knowledge is minimal). I'll look into a way of communicating ingestion progress, but for now maybe we can just show a spinner or something off to the side until the sidecar command returns.

Reply:
About the way to communicate progress, I think this is related to #33, we need to define the way we want to pass information between components, in this case the loading status/progress.

About how to do this piece, I guess it depends on when we want to run the ingestion process. For now, if we want to run it every time files are uploaded, that is the right place to do it. However, you need to call `runPDFIngestion` once files have been uploaded otherwise you will run into race conditions with the sidecar being called before the app directory receives the new files. I'm not a big fan of the `.then` chaining syntax and prefer the async/await, so you could write something like this:
```typescript
const handleChange = async (files: FileList) => {
    await uploadFiles(files);
    console.log('File uploaded with success');
    console.log(files);
    let updatedFiles = await readAllProjectFiles();
    setFiles(updatedFiles);

    await runPDFIngestion()
    console.log('PDFs ingested with success');
   updatedFiles = await readAllProjectFiles();
    setFiles(updatedFiles);
  };
```

Reply:
I would provide `runPDFIngestion` with both the current array of file paths uploaded, and also the base directory for the uploads. Both absolute paths to the filesystem. 

This would allow the ingestion to either read new files uploaded or the full list of files in the upload folder.

```js
// In filesystem.ts
export async function getUploadsDir() {
  return join(await getBaseDir(), UPLOADS_DIR);
}

// In FoldersView.tsx
const handleChange = async (files: FileList) => {
    await uploadFiles(files);
    console.log('File uploaded with success');
    console.log(files);
    let updatedFiles = await readAllProjectFiles();
    setFiles(updatedFiles);

    const uploadedFilesPath = updatedFiles.map(f => f.path)
    const uploadsDir = await getUploadsDir()
    await runPDFIngestion(uploadedFilesPath, uploadsDir)
    console.log('PDFs ingested with success');
    updatedFiles = await readAllProjectFiles();
    setFiles(updatedFiles);
  };
```